src/uproot/compression.py

```diff
@@ -115,6 +120,9 @@ def level(self):
         """
         The compression level: 0 is uncompressed, 1 is minimally compressed, and
         9 is maximally compressed.
+
+        This value to adapted to the ISAL compression levels if that library is used.
+        Note: with ISAL 0 is lowest compression, not uncompressed!
```

Comment:
With ISAL being strictly opt-in, its compression levels (only relevant for writing) can be different from ZLIB, including this fact that ISAL 0 is not uncompressed. Fortunately, it's also possible to set `compression=None` in [uproot.recreate](https://uproot.readthedocs.io/en/latest/uproot.writing.writable.recreate.html) and [uproot.update](https://uproot.readthedocs.io/en/latest/uproot.writing.writable.update.html) to make this distinction.

Reply:
> ISAL 0 is not uncompressed

I didn't know this! I suppose when no-compressor is also an option, it doesn't come up often.

Reply:
(I hadn't known it either, before reading @lgray's docstring.)

Actually, the `fCompress` variable encoded in ROOT files can't express "ZLIB level 0, distinct from no compression," either. (There is no [EAlgorithm](https://root.cern.ch/doc/master/structROOT_1_1RCompressionSetting_1_1EAlgorithm.html#ad5ed4ef81888f5406ae0f018ff45fd96) for "no compression," so if you want no compression, you set the algorithm to "ZLIB" and the level to "0".) Fortunately, that declaration is meaningless (aspirational?). Whether a block of data is compressed or not is determined by whether the compressed size is equal to the uncompressed size (because a ROOT-writer would never write a compressed block if it happens to be larger than an uncompressed one). For small and other not-very-compressible data, blocks might even alternate arbitrarily between compressed and uncompressed.

When writing, Uproot (not ROOT) has an API that can distinguish between intending to use such-and-such an algorithm with such-and-such a level or intending to not compress at all. So we can make all blocks uncompressed and report it in `fCompress` as "ZLIB level 0."

Reply:
> whether the compressed size is equal to the uncompressed size

So actually zlib(level=0) will never happen anyway, since they will actually always be slightly bigger than the original, with the addition of the block header.